flight/PiOS/Common/pios_dsm.c

```diff
@@ -316,9 +316,40 @@ int32_t PIOS_DSM_Init(uintptr_t *dsm_id,
 	/* Bind the configuration to the device instance */
 	dsm_dev->cfg = cfg;
 
+	uint8_t num_pulses = 0;
+
+	// check user settings to determine which resolution mode to use
+	// if invalid mode selected, bail
+	if (mode > HWSHARED_DSMXMODE_BIND10PULSES)
```

Comment:
I'm hoping @mlyle or @solidgoldbomb can chime in here. I see what you're doing and I wouldn't argue with the why: you're covering a case where the variable has potentially received a value not in the enum. In theory, the compiler should prevent this from occurring because you could not assign ```mode = 5;```, but I'm betting you could force it with ```mode = (HwSharedDSMxModeOptions) 5;```. And I'm sure you could force it even more easily with pointer foo.

So it makes a lot of sense to be on the lookout for this type of error, but this is the first time I've seen this range check applied so there's not exactly a long history of handling this. Even then, there is no guarantee that the enum is sequential, and that -1 is not a value. Without the ```default``` case, I don't see a way to *guarantee* goodness here, except via code review. And while I'm not a big fan of getting rid of ```default``` because the compiler catches the more likely error-- not updating all switches to handle a new enum-- even though it it could hide an insidious error, I'd love to hear from some Mike or Stacey.